### PR TITLE
Add New Endpoints ( SOFTWARE-4346 )

### PR DESCRIPTION
**Pull request by CannonLock**
Add the new endpoints and some tests to verify

### CODE REVIEW

**Review by brianhlin — COMMENTED** [review]
@matyasselmeci it took me a minute to swap this back in but looking at the endpoints, doesn't it make more sense to have `/<server-type>-<auth-config-type>`? In other words these would stay the same:

- `/cache-authfile`
- `/cache-authfile-public`

And we'd change the following:

- `/scitokens-cache` -> `/cache-scitokens`
- `/scitokens-origin` -> `/origin-scitokens`

**Comment by matyasselmeci** [comment]
Yes that does look better. Maybe we could even do
```
/cache/Authfile
/cache/Authfile-public
/cache/scitokens.conf
```

**Comment by brianhlin** [comment]
And `/origin/scitokens.conf`! @CannonLock could you go ahead and make these changes? Otherwise everything else looks reasonable to me

**Comment by matyasselmeci** [comment]
Right, `/origin/all of those`

**Comment by CannonLock** [comment]
@matyasselmeci @brianhlin 

To confirm there should be 6 new endpoints:

- /cache/Authfile
- /cache/Authfile-public
- /cache/scitokens.conf
- /origin/Authfile
- /origin/Authfile-public
- /origin/scitokens.conf

**Comment by CannonLock** [comment]
@matyasselmeci I have applied the changes and tests to accompany.

File: src/app.py [hunk 1]
```diff
@@ -337,7 +351,7 @@ def oasis_managers():
 def _get_cache_authfile(public_only):
     if not stashcache:
         return Response("Can't get authfile: stashcache module unavailable", status=503)
-    cache_fqdn = request.args.get("cache_fqdn")
+    cache_fqdn = request.args.get("cache_fqdn") if request.args.get("cache_fqdn") else request.args.get("fqdn")
```
Comment (matyasselmeci) [inline src/app.py:354]:
```suggestion
    cache_fqdn = request.args.get("fqdn") if request.args.get("fqdn") else request.args.get("cache_fqdn")
```

File: src/app.py [hunk 2]
```diff
@@ -394,6 +408,53 @@ def _get_origin_authfile(public_only):
     return Response(auth, mimetype="text/plain")
 
 
+def _get_scitoken_file(fqdn, get_scitoken_function):
+
+    if not stashcache:
+        return Response("Can't get scitokens config: stashcache module unavailable", status=503)
+
+    if not fqdn:
+        return Response("FQDN of cache or origin server required in the 'cache_fqdn' or 'origin_fqdn' argument", status=400)
```
Comment (matyasselmeci) [inline src/app.py:417]:
```suggestion
        return Response("FQDN of cache or origin server required in the 'fqdn' argument", status=400)
```

File: src/tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            assert previous_endpoint.status_code == current_endpoint.status_code
+            assert previous_endpoint.data == current_endpoint.data
+
+    def test_cache_authfile_public(self, client: flask.Flask):
+        resources = client.get('/miscresource/json').json
+        for resource in resources.values():
+
+            resource_fqdn = resource["FQDN"]
+            previous_endpoint = client.get(f"/cache/Authfile-public?fqdn={resource_fqdn}")
+            current_endpoint = client.get(f"/stashcache/authfile-public?cache_fqdn={resource_fqdn}")
```
Comment (matyasselmeci) [inline src/tests/test_api.py:84]:
```suggestion
            current_endpoint = client.get(f"/cache/Authfile-public?fqdn={resource_fqdn}")
            previous_endpoint = client.get(f"/stashcache/authfile-public?cache_fqdn={resource_fqdn}")
```

File: src/tests/test_api.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            assert previous_endpoint.status_code == current_endpoint.status_code
+            assert previous_endpoint.data == current_endpoint.data
+
+    def test_origin_authfile(self, client: flask.Flask):
+        resources = client.get('/miscresource/json').json
+        for resource in resources.values():
+
+            resource_fqdn = resource["FQDN"]
+            previous_endpoint = client.get(f"/origin/Authfile?fqdn={resource_fqdn}")
+            current_endpoint = client.get(f"/stashcache/origin-authfile?fqdn={resource_fqdn}")
```
Comment (matyasselmeci) [inline src/tests/test_api.py:95]:
```suggestion
            current_endpoint = client.get(f"/origin/Authfile?fqdn={resource_fqdn}")
            previous_endpoint = client.get(f"/stashcache/origin-authfile?fqdn={resource_fqdn}")
```

File: src/tests/test_api.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            assert previous_endpoint.status_code == current_endpoint.status_code
+            assert previous_endpoint.data == current_endpoint.data
+
+    def test_origin_authfile_public(self, client: flask.Flask):
+        resources = client.get('/miscresource/json').json
+        for resource in resources.values():
+
+            resource_fqdn = resource["FQDN"]
+            previous_endpoint = client.get(f"/origin/Authfile-public?fqdn={resource_fqdn}")
+            current_endpoint = client.get(f"/stashcache/origin-authfile-public?fqdn={resource_fqdn}")
```
Comment (matyasselmeci) [inline src/tests/test_api.py:106]:
```suggestion
            current_endpoint = client.get(f"/origin/Authfile-public?fqdn={resource_fqdn}")
            previous_endpoint = client.get(f"/stashcache/origin-authfile-public?fqdn={resource_fqdn}")
```

**Review by matyasselmeci — CHANGES_REQUESTED** [review]
Mostly just some text that needs to be swapped; otherwise it's fine. See comments.

**Review by matyasselmeci — APPROVED** [review]
LGTM.

**Comment by matyasselmeci** [comment]
You should merge changes from master and then I can merge this PR.

**Comment by CannonLock** [comment]
@matyasselmeci Done

**Pull request merged**